### PR TITLE
Updates upward-security-headers peer dependencies

### PR DESCRIPTION
**Pull request by supernova-at**
## Description

This PR updates the `upward-security-headers` package's `peerDependencies`.

Specifically:
1. `@magento/pwa-buildpack`: Update matches the version in [v7.0.0](https://github.com/magento/pwa-studio/blob/v7.0.0/packages/pwa-buildpack/package.json#L3)
1. `@magento/venia-ui`: Update matches the version in [v7.0.0](https://github.com/magento/pwa-studio/blob/v7.0.0/packages/venia-ui/package.json#L3)
1. `webpack`: Unchanged. The current version matches the version of webpack as specified in [venia-concept](https://github.com/magento/pwa-studio/blob/v7.0.0/packages/venia-concept/package.json#L114)

## Related Issue

Closes PWA-820.

### Verification Stakeholders

@dpatil-magento @zetlen 

### Verification Steps

1. Security score for webpagetest should be A+.

## Checklist

- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2605.pwa-venia.com : LH Performance Expected 0.85 Actual 0.34, LH Accessibility Expected 1 Actual 0.97, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 39.333333333333
https://pr-2605.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2605.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.41, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 50

File: packages/extensions/upward-security-headers/package.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-  "license": "(OSL-3.0 OR AFL-3.0)",
-  "peerDependencies": {
-    "@magento/pwa-buildpack": "^5.1.1",
-    "@magento/venia-ui": "^3.0.0",
-    "webpack": "~4.38.0"
-  },
-  "pwa-studio": {
-    "targets": {
-      "intercept": "./intercept"
+    "name": "@magento/upward-security-headers",
```
Comment (tjwiebell) [inline packages/extensions/upward-security-headers/package.json:2]: Gotta watch editor auto formatting of `package.json` files, they don't seem to pick up on `.editorconfig` in the mono root. Can you fix the white space on this file?

File: packages/extensions/upward-security-headers/package.json [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+    "main": "intercept.js",
+    "scripts": {
+        "clean": " "
+    },
+    "repository": "github:magento/pwa-studio",
+    "author": "Magento Commerce",
+    "license": "(OSL-3.0 OR AFL-3.0)",
+    "peerDependencies": {
+        "@magento/pwa-buildpack": "^6.0.0",
+        "@magento/venia-ui": "^4.0.0",
```
Comment (tjwiebell) [inline packages/extensions/upward-security-headers/package.json:17]: Also weird, your update didn't use the `save-prefix` in `.yarnrc`; we're still on the tilde train.

```suggestion
        "@magento/pwa-buildpack": "~6.0.0",
        "@magento/venia-ui": "~4.0.0",
```

File: packages/extensions/upward-security-headers/package.json (hunk 1 above)
Comment (supernova-at) [reply packages/extensions/upward-security-headers/package.json]: Oops, thanks. Fixed in https://github.com/magento/pwa-studio/pull/2605/commits/5ee885fcfe5ab2c4c7173bb268f9b4ea065e3ec0.

File: packages/extensions/upward-security-headers/package.json (hunk 2 above)
Comment (supernova-at) [reply packages/extensions/upward-security-headers/package.json]: ALLLLLLLLrighty then. https://github.com/magento/pwa-studio/pull/2605/commits/5ee885fcfe5ab2c4c7173bb268f9b4ea065e3ec0.

**Pull request merged**